quantum/split_common/transactions.c

```diff
@@ -264,8 +264,20 @@ static bool layer_state_handlers_master(matrix_row_t master_matrix[], matrix_row
 }
 
 static void layer_state_handlers_slave(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) {
-    layer_state         = split_shmem->layers.layer_state;
-    default_layer_state = split_shmem->layers.default_layer_state;
+    if (layer_state != split_shmem->layers.layer_state) {
+        layer_state         = split_shmem->layers.layer_state;
+
+#if defined(SPLIT_LAYER_STATE_SET)
+        layer_state         = layer_state_set_kb(layer_state);
+#endif 
+    }
+    if (default_layer_state != split_shmem->layers.default_layer_state) {
+        default_layer_state = split_shmem->layers.default_layer_state;
+
+#if defined(SPLIT_LAYER_STATE_SET)
+        default_layer_state = default_layer_state_set_kb(default_layer_state);
+#endif
+    }
```

Comment:
Would rather have it just set, no mess, no fuss, just called. 

```suggestion
    if (layer_state != split_shmem->layers.layer_state) {
        layer_state_set(split_shmem->layers.layer_state);
    }
    if (default_layer_state != split_shmem->layers.default_layer_state) {
       default_layer_set(split_shmem->layers.default_layer_state);
    }
```

this should only call the function if the states are different, so it should minimize the frequency that they are called.

And it uses the lowest level version to do this, so should be safer, in theory, and supports debugging, if that's enabled. 


Otherwise:
```suggestion
    if (layer_state != split_shmem->layers.layer_state) {
#if defined(SPLIT_LAYER_STATE_SET)
        layer_state_set(split_shmem->layers.layer_state);
#else
        layer_state         = split_shmem->layers.layer_state;
#endif 
    }
    if (default_layer_state != split_shmem->layers.default_layer_state) {
#if defined(SPLIT_LAYER_STATE_SET)
        default_layer_set(split_shmem->layers.default_layer_state);
#else
        default_layer_state = split_shmem->layers.default_layer_state;
#endif
    }
```

Reply:
> Would rather have it just set, no mess, no fuss, just called.

That works for me certainly. I don't have much experience with QMK to say whether it would break other usecases, but it doesn't seem like it should.